.github/workflows/test.yaml

```diff
@@ -4,36 +4,65 @@ on:
     branches:
       - main
   pull_request:
+
+defaults:
+  run:
+    shell: bash
+
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+          - windows-latest
+          - macos-latest
     steps:
       - uses: actions/checkout@v2
       - run: tools/build.sh
       - run: git diff --exit-code
       - uses: actions/upload-artifact@v2
         with:
-          name: build
+          name: build-${{ matrix.os }}
           path: target/release/turtle
   unit_test:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+          - windows-latest
+          - macos-latest
     steps:
       - uses: actions/checkout@v2
       - run: cargo test
   integration_test:
     needs: build
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+          - windows-latest
+          - macos-latest
     steps:
       - uses: actions/checkout@v2
       - uses: ruby/setup-ruby@v1
       - uses: actions/download-artifact@v2
         with:
-          name: build
+          name: build-${{ matrix.os }}
           path: target/release
       - run: chmod +x target/release/turtle
       - run: tools/integration_test.sh
   integration_test_test:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+          - windows-latest
```

Comment:
I'm not sure what is the best or standard way to install `ninja` on Windows. Homebrew is not available there, which is failing this test...

Reply:
Fixed in https://github.com/raviqqe/turtle-build/pull/149/commits/1b79011846dfb2555c3d5e7c0a10c5aca48ff6bb